components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java

```diff
@@ -114,6 +152,8 @@ public void init(SynapseEnvironment synapseEnvironment) {
         try {
             authenticator = new OAuthAuthenticator();
             saml2Authenticator = new SAML2Authenticator();
+            webApp = new WebApp();
+            initJWTCapabilities();
         } catch (Exception e) {
             // Just throw it here - Synapse will handle it
             throw new SynapseException("Error while initializing SAML or OAuth authenticator");
```

Comment:
We should not catch generic exceptions. Here we have used generic exception(Exception e)